### DISCUSSION TITLE
OAuth2 - Unique email address required for user registration

### DISCUSSION THREAD

**Discussion by bsuchorowski** [requarks/wiki#6635] [Help / Questions]
Hi,
I have a similar question to the one in #6225. We use "Generic OpenID Connect / OAuth2" to register/authenticate users via our own Identity Server (running on Duende). Everything works fine until we face a user with the same email. It is often in our system that users have muliple accounts with different usernames but same email.

We thought that we could prepare an additional property "fake_email" with valid but fake & unique email address that we then next put in "Email Claim" in "Generic OpenID Connect / OAuth2" configuration. Unfortunately this doesn't work and we are still getting UK violation on a database level.

As a workaround, we modified our Identity Server code to put this fake & unique email into default email property. It works but this way we lost user email information in case any other system would like to integrate and make use of it.

Did we misused "Email Claim" and did not understand what is that for?

**Comment by NGPixel** [comment]
Make sure your identity server doesn't return an "emails" array property in addition to the fake_email property, as Wiki.js will always use the emails array first (and pick the first email from that array), then fallback on the email property (which is set to the email claim).
